client/js/views/Home.vue

```diff
@@ -119,8 +119,18 @@ export default {
       return this.$store.state.gameList.games;
     },
   },
-  mounted() {
-    this.$store.dispatch('requestGameList');
+  async mounted() {
+    // Leave any existing lobbies before getting the game list
+    // Need to make sure we do these in order so the order of operations is:
+    //     leave -> list
+    try {
+      await this.$store.dispatch('requestLeaveLobby');
+    } catch (err) {
+      // A user may not actually be in a lobby, so we need to
+      // swallow this error so the component doesn't error out
+      // TODO: Improve this logic with an isInLobby helper
```

Comment:
We may also want to make the backend more robust here as a part of that cleanup. I think it'd make sense for the backend to just silently 200 you if you request to leave a lobby when you're not in one. Then we wouldn't expect an error here anyway